src/analysis/utils/gce-machines.ts

```diff
@@ -28,8 +34,13 @@ export const machineTypes = [
 ];
 
 // returns true if machineTypeA machine type is smaller than machineTypeB
-export const isMachineTypeSmaller = (machineTypeA, machineTypeB) =>
-  _.find({ name: machineTypeA }, machineTypes).cpu < _.find({ name: machineTypeB }, machineTypes).cpu;
+export const isMachineTypeDifferent = (machineTypeA: string, machineTypeB: string): boolean => {
```

Comment:
Nit, but do you want to modify the comment above?

Also was there some business logic to only look at smaller machine sizes before? aka only a smaller type would trigger a runtime creation?

Reply:
this was a bug that resulted in errors in the console (currently in dev) when you switch the compute profile while not having a runtime. All you have to do is go to a workspace with no runtime, and switch to a `single spark node` and look in the console. 

I wanted to avoid modifying the business logic and touching anything in `GcpComputeModal` as much as possible but added `undefined` handling to this function. It still checks `isMachineTypeSmaller` so I'll revert the name, but modify the comment to capture the behavior if one of the args is undefined.